src/main/java/se/knowit/bookitevent/service/map/EventServiceMapImpl.java

```diff
@@ -24,15 +24,48 @@ public EventServiceMapImpl() {
 
 	@Override
 	public Event save(Event incomingEvent) {
-		Event event = Objects.requireNonNull(incomingEvent, "Event argument must not be null");
-		
+		Event event = ensureEventIsValidOrThrowException(incomingEvent);
+		assignRequiredIds(event);
+		map.put(event.getId(), event);
+		return event;
+	}

```

Comment:
I think it is nice if the private methods are structured in the same order as they are called (at least if they are only used once).

```
private Event ensureEventIsValidOrThrowException(Event incomingEvent)
private void ensureNameIsSet(Event event)
private void ensureStartTimeIsSet(Event event)

private void assignRequiredIds(Event event)
private void assignPersistenceIdIfNotSet(Event event)
private void assignEventIdIfNotSet(Event event)
```

Reply:
Pushed changes